landoui/treestatus.py

```diff
@@ -211,15 +211,25 @@ def treestatus_tree(tree: str):
     """Display the log of statuses for an individual tree."""
     api = TreestatusAPI.from_environment()
 
-    logs_response = api.request("GET", f"trees/{tree}/logs")
+    try:
+        logs_response = api.request("GET", f"trees/{tree}/logs")
+    except LandoAPIError as exc:
+        if not exc.detail:
+            raise
+
+        error = f"Error received from LandoAPI: {exc.detail}"
+        logger.error(error)
+        flash(error, error)
+        return redirect(request.referrer, code=exc.status)
+
     logs = logs_response.get("result")
     if not logs:
         logger.error(f"Could not retrieve logs for tree {tree}.")
         flash(
             f"Could not retrieve status logs for {tree} from Lando, try again later.",
             "error",
         )
-        return redirect(request.referrer)
+        return redirect(request.referrer, code=404)
```

Comment:
Is it not possible to have no logs for a tree? Say a new tree?

Reply:
Yes, a new tree would have no log entries. However failure to retrieve any logs for a tree that should have them is a problem. I think this would be better to resolve by adding an initial log entry for a new tree.

Reply:
This code in its current state is incorrect then, since the error being logged in not necessarily accurate (i.e., "Could not retrieve logs for tree {tree}.").

The behaviour may also be confusing (i.e., returning a 404 for this particular situation). I.e., differentiating between a tree with no logs and a failure to fetch logs would be good here.

Reply:
Given any new tree will have no log entries for ~1 day at most, it's not worth the effort. Adding a log entry when a new tree is created keeps things simpler, since the "initial commit" for a tree should create a log entry.

Reply:
This is a matter of correctness/accuracy. At least this should not be a 404, and this change in itself is out of scope for this PR. Regardless if we added an initial log entry in Lando for new trees, this would still be not correct.

Reply:
@zzzeid removed the `code=404`.